lib/profile/WP_Auth0_Profile_Change_Password.php

```diff
@@ -44,14 +52,17 @@ public function init() {
 	public function validate_new_password( $errors, $user ) {
 
 		// Exit if we're not changing the password.
-		if ( empty( $_POST['pass1'] ) ) {
+		// The pass1 key is for core WP, password_1 is WooCommerce.
+		if ( empty( $_POST['pass1'] ) && empty( $_POST['password_1'] ) ) {
 			return false;
 		}
-		$new_password = $_POST['pass1'];
+
+		$field_name   = ! empty( $_POST['pass1'] ) ? 'pass1' : 'password_1';
+		$new_password = $_POST[ $field_name ];
 
 		if ( isset( $_POST['user_id'] ) ) {
 			$wp_user_id = absint( $_POST['user_id'] );
-		} elseif ( is_object( $user ) && $user instanceof WP_User ) {
+		} elseif ( is_object( $user ) && ! empty( $user->ID ) ) {
```

Comment:
will this second condition have the same effect as the one that was here previously?

Reply:
No, totally different but the hook we're attaching to does not use `WP_User` and, really, all we need is the user ID.